libs/sdk-core/src/models.rs

```diff
@@ -132,63 +135,113 @@ pub struct ReverseSwapPairInfo {
 }
 
 /// Details of past or ongoing reverse swaps, as stored in the Breez local DB
-#[derive(Clone, Debug, Serialize)]
+#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
 pub struct ReverseSwapInfo {
-    // static immutable data
-    pub claim_address: String,
-    pub lockup_address: String,
+    /// The reverse swap ID, as reported by the Boltz API in case of a successful creation
+    pub id: String,
 
-    // dynamic data
-    pub status: ReverseSwapStatus,
+    pub created_at: i64,
+
+    // Locally generated fields
+    pub local_preimage: Vec<u8>,
+    pub local_private_key: Vec<u8>,
+
+    pub destination_address: String,
+
+    // Other fields from the Boltz API response
+    /// Reverse swap status, as reported by the Boltz API. Automatically updated on every new block.
+    pub boltz_api_status: BoltzApiReverseSwapStatus,
+
+    // Only available in the first response from Boltz API:
+    pub lockup_address: String,
+    pub hodl_bolt11: String,
+    pub onchain_amount_sat: u64,
+    pub redeem_script: String,
+}
+
+impl ReverseSwapInfo {
+    /// Dynamically determine the status, based on the currently known Boltz API status for this reverse swap
+    pub fn status(&self) -> ReverseSwapStatus {
+        match &self.boltz_api_status {
+            BoltzApiReverseSwapStatus::SwapCreated => ReverseSwapStatus::Initial,
```

Comment:
Boltz uses very detailed status but I am not sure we need/want to expose that to the user. IMO we can map these statuses to :
Initial - created in the db and lock tx is not yet in mempool
InProgress - lock tx in mempool
Completed - successfully completed
Expired/Invalidated -   swap is not usable anymore and shouldn't be monitored.

Could be other options as well as long as they are simplified and not exposing the internals @yaslama @ok300 what do you think?

Reply:
@ok300 on second thought I now see that the ReverseSwapStatus is not exposed to the user so why do we need it at all. 
Can't we use internally the boltz status?

Reply:
This is addressed on two levels:

First, the possible values for `ReverseSwapStatus` were simplified.

Second, the `BreezServices` doesn't return the full `ReverseSwapInfo` to the user, containing internals like the HODL invoice or the redeem script. Instead, a stripped-down version of that is returned in 7f3d5a25fbc0853722f407b4b100f11e03dbe30d

https://github.com/breez/breez-sdk/blob/7f3d5a25fbc0853722f407b4b100f11e03dbe30d/libs/sdk-core/src/models.rs#L311-L317